test/objectFifo-stateful-transform/nested_loop_test.mlir

```diff
@@ -0,0 +1,145 @@
+//===- nested_loop_test.mlir -----------------------------------------*- MLIR -*-===//
```

Comment:
It would be easier to understand the new functionality with a simpler test case, maybe with less object FIFOs (we're specifically looking at the ones for tile_1_2) and without all the toStream and fromStream attributes. It's a bit hard to read the test in its current state and I don't think all the attributes are needed to showcase the loop unrolling. Could you please make that change?

Reply:
Hi @AndraBisca - I've made the changes. Could you re-review?

Also, I'm using ` ./utils/build-mlir-aie.sh llvm/build` to build MLIR-AIE and using the same command for running the lit test.

Strangely it's passing locally - any pointers?

Reply:
Thank you for making the changes! It's already so much easier to read and reason on! :) 

Did you also update your environment variables with `./utils/env_setup.sh`?

I will pull your branch locally and try it out.

Reply:
Hi @AndraBisca - yes, I did do that.

Also, the run command I use is `./build/bin/aie-opt test/objectFifo-stateful-transform/nested_loop_test.mlir --aie-objectFifo-stateful-transform` - which works locally for me. So, I'm not really depending on `aie-opt` to be added as environment variable anyway.

Please let me know if you are able to reproduce the error - I'll try following the CI workflow to see if I can replicate it at my end. My hunch is that this might be because of Assert/Release enchilada.

Reply:
I have pulled the branch and was unable to reproduce the error, on my end the output is as expected in the FileCheck.

Reply:
Oh okay. Thanks for trying @AndraBisca !

@makslevental @fifield would you happen to know the reason behind this issue and a way to reproduce the error that the CI is reporting locally?

I'm following the [installation instructions here](https://github.com/Xilinx/mlir-aie/blob/main/docs/Building.md#building-on-x86) and invoking the .sh scripts

Reply:
@AndraBisca I was able to replicate the issue and create a fix. Could you have another look?